### PR TITLE
Bump Go to v1.22

### PR DESCRIPTION
**Pull request by emcfarlane**
Increases the min go version to v1.20 and the current to v1.22. Also bumps the golinter to v1.56.2 and fixes the issues found.

### CODE REVIEW

**Comment by emcfarlane** [comment]
Settings require updating the merge constraints.

File: protocol_http.go [hunk 1]
```diff
@@ -33,6 +33,10 @@ import (
 	"google.golang.org/protobuf/types/known/anypb"
 )
 
+const (
+	contentApplicationPrefix = "application/"
+)
```
Comment (jhump) [inline protocol_http.go:38]: This belongs with the other constants over in `protocol_connect.go`: it's not used at all in this file but is used numerous times there.

Also, perhaps name it `contentConnectUnaryPrefix` and change the other one to `contentConnectStreamPrefix`?

**Review by jhump — APPROVED** [review]
Just one thing to change, then LGTM.

File: protocol_http.go (hunk 1 above)
Comment (emcfarlane) [reply protocol_http.go:38]: It's used in the rest protocol which is what caused the split. Didn't want the constant to have connect annotation when its a HTTP thing. wdyt?

Comment (emcfarlane) [reply protocol_http.go]: Separated out to separate const for each.

**Pull request merged**